src/controllers/engine/controllerengine.cpp

```diff
@@ -79,39 +80,50 @@ Input:   -
 Output:  -
 -------- ------------------------------------------------------ */
 void ControllerEngine::callFunctionOnObjects(QList<QString> scriptFunctionPrefixes,
-                                             const QString& function, QScriptValueList args) {
-    const QScriptValue global = m_pEngine->globalObject();
+                                             const QString& function, QJSValueList args) {
+    const QJSValue global = m_pScriptEngine->globalObject();
 
     for (const QString& prefixName : scriptFunctionPrefixes) {
-        QScriptValue prefix = global.property(prefixName);
-        if (!prefix.isValid() || !prefix.isObject()) {
+        QJSValue prefix = global.property(prefixName);
+        if (!prefix.isObject()) {
             qWarning() << "ControllerEngine: No" << prefixName << "object in script";
             continue;
         }
 
-        QScriptValue init = prefix.property(function);
-        if (!init.isValid() || !init.isFunction()) {
+        QJSValue init = prefix.property(function);
+        if (!init.isCallable()) {
             qWarning() << "ControllerEngine:" << prefixName << "has no" << function << " method";
             continue;
         }
         controllerDebug("ControllerEngine: Executing" << prefixName << "." << function);
-        init.call(prefix, args);
+        init.callWithInstance(prefix, args);
     }
 }
 
+QJSValue ControllerEngine::byteArrayToScriptValue(const QByteArray byteArray) {
+    // The QJSEngine converts the QByteArray to an ArrayBuffer object.
+    QJSValue arrayBuffer = m_pScriptEngine->toScriptValue(byteArray);
+    // We convert the ArrayBuffer to a Uint8 typed array so we can access its bytes
+    // with the [] operator.
+    QJSValue m_byteArrayToScriptValueJSFunction = evaluateCodeString("(function(arg1) { return new Uint8Array(arg1) })");
```

Comment:
Evaluating this code string every time input is received is inefficient. How about caching this in a private member variable in the ControllerEngine constructor?

Reply:
Oh, it already is initialized in `initializeScriptEngine`, but this local variable has the same name and is shadowing the class member. I think you can simply delete this line.